src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -332,36 +365,82 @@ InvokerFactory MakeMlirWrWInvokerFactory(const ConvolutionContext& ctx)
                            weights_strides,
                            out_dims,
                            out_strides);
-    MlirConvArgs args =
-        MakeMlirConvArgs(in_dims, in_strides, weights_dims, weights_strides, out_dims, out_strides);
+    MlirConvArgs args = MakeMlirConvArgs(in_dims,
```

Comment:
Looks like you need an if condition here: Based on whether a workspace is needed, invoke either `MakeMlirConvArgs()` or `MakeMlirConvArgsWithWorkspace()`.

Reply:
Same as above. I'd like to avoid duplicated logic in boilerplate codes so I'm adding this optional `bool` field.

Reply:
You can have `MakeMlirConvArgsWithWorkspace()` call `MakeMlirConvArgs()` that in the end result each function is simplified.

Reply:
But then the workspace arg in `MlirConvArgs` will be uninitialized. 

This may not be a problem since we are not effectively using all fields of a memref but I prefer this to be addressed to be conservative.

Reply:
@jerryyin I think I need to understand the architecture between a solver / an invoker / a ConvolutionContext better to see how to derive a better solution.

In a `ConvMlirIgemmWrWXdlops` solver, given a `ConvolutionContext`, we can use `MiirGetWorkspaceSize` to compute the size of workspace needed. How do we achieve the same here at `MakeMlirWrWInvokerFactor`?

Reply:
@jerryyin So my main question is:

How to derive if a workspace is needed within this couple of lines?

```
InvokerFactory MakeMlirWrWInvokerFactory(const ConvolutionContext& ctx)
{
    assert((ctx.direction.IsBackwardWrW()));

    std::vector<size_t> in_dims, in_strides;
    std::vector<size_t> weights_dims, weights_strides;
    std::vector<size_t> out_dims, out_strides;
    ComputeMlirDimsStrides(ctx.conv_problem,
                           in_dims,
                           in_strides,
                           weights_dims,
                           weights_strides,
                           out_dims,
                           out_strides);
```

Can we invoke `ConvMlirIgemmWrWXdlops::GetWorkspaceSize` here? But `ConvMlirIgemmWrWXdlops::GetWorkspaceSize` is a member function, not a class function, and an invoker does not possess a reference to a solver.

Reply:
@whchung `MakeMlirWrWInvokerFactory()` is called from `GetSolution()`. Compute workspace size in `GetSolution()` and pass it to `MakeMlirWrWInvokerFactory()` as a parameter.

[Informative] BTW you can instantiate Solver wherever you want and call its methods provided that you have necessary parameters on hand. Example: `ConvMlirIgemmWrWXdlops{}.GetWorkspaceSize(ctx)`.

Reply:
Per discussion with @jerryyin, I'll add some comments to explain why the implementation is done in this fashion. Meanwhile, we have reached a consensus that the existing implementation is okay for the time being.